templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -41,6 +41,9 @@ contents:
             old_pids=$(pidof haproxy)
             if [ -n "$old_pids" ]; then
                 /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid -x /var/lib/haproxy/run/haproxy.sock -sf $old_pids &
+                #There seems to be some cases where HAProxy doesn't drain properly. 
+                #To handle that case, SIGTERM signal being sent to old HAProxy processes which haven't terminated after few seconds.
+                sleep 5 && bash -c "echo $old_pids|xargs kill " &
```

Comment:
is `$old_pids` multiline? Do we really need xargs?

Reply:
@celebdor , Well, that's an interesting point, my original code didn't include the xargs.
I tested my code by forcing haproxy-monitor to send 'reload' request every few seconds, and sometimes $old_pids was multiline. so to be on the safe side I decided to use xargs.

Reply:
That's odd. pidof defaults to delimiting multiple pids using whitespace, not a newline. When old_pids was multiline was it multiple pids or some other unexpected output?

Reply:
I see there is a -st option to haproxy that is the corresponding SIGTERM command to the -sf one used above. Maybe we could just use that so we have consistency in how we're signalling?

Reply:
The 'old' Haproxy not being deleted since there are still open connections through these processes. 

Using '-st' means that processes will be terminated immediately, and that may break ongoing sessions, while '-sf' terminated the processes only after all the ongoing sessions completed. So, I don't think that '-st' is a good option here.
 
I continued investigating this issue, and it seems that 'old' processes not being deleted (when '-sf' is used) because of the long timeouts we set in [1], I think that decreasing these timeout values is the right way to solve this bug.
I'll update this PR 

[1] https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-haproxy-haproxy.yaml#L15-#L17

Reply:
Right, I wasn't suggesting that we replace -sf with -st, but use -st to send the sigterm here.

We may need to be careful about decreasing timeouts. I've seen issues in the past where applications that open persistent connections can be broken by haproxy timeouts. I'm guessing that's why they are so long now.

Reply:
A. Yes, I know that we should be sensitive with HAProxy timeout values, but I'm not sure that current timeouts values are correct. 
I suggested using the default values used by OCP HAProxy routers [1]. 

B. Not sure that I understood your point with the '-st', could you elaborate about that?

[1] https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template#L114-#L123